### PR TITLE
Restore building web platform by enclosing resolve_function_signature.

### LINKED ISSUE #58852: No member named 'default_arg_values' in 'GDScriptParser::FunctionNode'
### Godot version

ffd64505c51cb9734ad68f52277c807a9b759338

### System information

Fedora 35, Nvidia 3000 series

### Issue description

Can't build web platform.

```
modules/gdscript/gdscript_analyzer.cpp:1172:33: error: no member named 'default_arg_values' in 'GDScriptParser::FunctionNode'
                        valid = valid && p_function->default_arg_values.size() >= default_par_count + par_count_diff;
```

### Steps to reproduce

```
scons werror=no platform=javascript target=release_debug -j`nproc` use_lto=no deprecated=no use_llvm=yes builtin_freetype=yes initial_memory=256 CCFLAGS='-fno-stack-protector -fno-exceptions' custom_modules=../godot-modules-groups
```

### Minimal reproduction project

Use the above scons command.

### PR DESCRIPTION
**Pull request by fire**
Fixes: https://github.com/godotengine/godot/issues/58852

Not my area. Please review or revise.

### CODE REVIEW

File: modules/gdscript/gdscript_parser.h [hunk 1]
```diff
@@ -733,10 +733,9 @@ class GDScriptParser {
 		Multiplayer::RPCConfig rpc_config;
 		MethodInfo info;
 		LambdaNode *source_lambda = nullptr;
-#ifdef TOOLS_ENABLED
+
 		Vector<Variant> default_arg_values;
 		String doc_description;
```
Comment (akien-mga) [inline modules/gdscript/gdscript_parser.h:738]: I think this can be kept tools only, no?

Comment (fire) [reply modules/gdscript/gdscript_parser.h:738]: The `p_function->default_arg_values.size()` in gdscript_analyzer.cpp is protected by a is debug.

Comment (fire) [reply modules/gdscript/gdscript_parser.h:738]: Maybe this should be a is debug defined too

Comment (akien-mga) [reply modules/gdscript/gdscript_parser.h:738]: The proper fix is likely to leave this unchanged and make *this* `TOOLS_ENABLED`, like all other code accessing `default_arg_values`: https://github.com/godotengine/godot/pull/58835/files#diff-e01550d654cd3b28e2c1abaa393c95a08008ea22c5860cbd034b515b84635237R1157

Comment (fire) [reply modules/gdscript/gdscript_parser.h]: This change I made looks ok.

File: modules/gdscript/gdscript_analyzer.cpp [hunk 2]
```diff
@@ -1154,7 +1154,7 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 		GDScriptParser::DataType return_type = resolve_datatype(p_function->return_type);
 		p_function->set_datatype(return_type);
 
-#ifdef DEBUG_ENABLED
+#if defined(DEBUG_ENABLED) && defined(TOOLS_ENABLED)
```
Comment (akien-mga) [inline modules/gdscript/gdscript_analyzer.cpp:1157]: `TOOLS_ENABLED` implies `DEBUG_ENABLED`, so no need for both. All other code that uses `default_arg_values` uses `#ifdef TOOLS_ENABLED` so there's no reason to diverge here (or all other call sites should be changed too).

The commit message and PR title should also be changed.

Comment (fire) [reply modules/gdscript/gdscript_analyzer.cpp]: Did the shortest change and renamed pr.

Feel free to change the title or design. Good night.

Comment (fire) [reply modules/gdscript/gdscript_analyzer.cpp]: The previous code used.. #ifdef ... ahhh

**Review by akien-mga — APPROVED** [review]

**Pull request merged**